src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java

```diff
@@ -0,0 +1,1202 @@
+        for (AchIncomeFileTransactionReference fileTransactionReference : fileTransactionReferences) {
+            if (CuFPConstants.AchIncomeFileTransactionReference.REF_REFERENCE_TYPE_TN.equals(fileTransactionReference.getType())) {
+                notes.append("TRANSACTION NUMBER: ");
+                notes.append(fileTransactionReference.getValue() + "\n");
+            }
+
+            if (CuFPConstants.AchIncomeFileTransactionReference.REF_REFERENCE_TYPE_CT.equals(fileTransactionReference.getType())) {
+                notes.append("CONTRACT NUMBER: ");
+                notes.append(fileTransactionReference.getValue() + "\n");
+            }
+
+            if (CuFPConstants.AchIncomeFileTransactionReference.REF_REFERENCE_TYPE_VV.equals(fileTransactionReference.getType())) {
+                notes.append("VOUCHER NUMBER: ");
+                notes.append(fileTransactionReference.getValue() + "\n");
+            }
+        }
+
+        List<AchIncomeFileTransactionDateTime> transactionDateTimes = achIncomeFileTransaction.getDateTimes();
+        for (AchIncomeFileTransactionDateTime transactionDateTime : transactionDateTimes) {
+            if (CuFPConstants.AchIncomeFileTransactionDateTime.DTM_DATE_TYPE_097.equals(transactionDateTime.getType())) {
+                notes.append("TRANSACTION CREATION: ");
+                notes.append(transactionDateTime.getDateTime() + "\n");
+            }
+        }
+
+        String payerNoteText = "";
+        String receiverNoteText = "";
+
+        List<AchIncomeFileTransactionPayerOrPayeeName> payerOrPayeeNames = achIncomeFileTransaction.getPayerOrPayees();
+        for (AchIncomeFileTransactionPayerOrPayeeName payerOrPayeeName : payerOrPayeeNames) {
+            if (CuFPConstants.AchIncomeFileTransactionPayerOrPayeeName.PAYER_TYPE_PE.equals(payerOrPayeeName.getType())) {
+                payerNoteText = payerOrPayeeName.getName();
+                if (payerOrPayeeName.getIdCode() != null &&
+                        payerOrPayeeName.getIdQualifier() != null) {
+                    payerNoteText = payerNoteText + " " + payerOrPayeeName.getIdCode() + ":" + payerOrPayeeName.getIdQualifier();
+                }
+            }
+
+            if (CuFPConstants.AchIncomeFileTransactionPayerOrPayeeName.PAYER_TYPE_PR.equals(payerOrPayeeName.getType())) {
+                receiverNoteText = payerOrPayeeName.getName();
+            }
+        }
+
+        if (StringUtils.isBlank(payerNoteText) && achIncomeFileTransaction.getPremiumAdminsContact() != null) {
+            payerNoteText = achIncomeFileTransaction.getPremiumAdminsContact().getName();
+        }
+
+        if (StringUtils.isBlank(receiverNoteText) && achIncomeFileTransaction.getPremiumReceiverName() != null) {
+            receiverNoteText = achIncomeFileTransaction.getPremiumReceiverName().getName();
+        }
+
+        if (StringUtils.isNotBlank(payerNoteText)) {
+            notes.append("RECEIVER: ");
```

Comment:
Not sure since I didn't actually write the original code :-), but it does seem confusing. I'll look further and switch it if that makes sense, thanks.